lib/pbench/test/unit/common/test_config.py

```diff
@@ -29,224 +29,119 @@ def get(self, section: str, option: str, fallback=None) -> str:
             else:
                 return fallback
-        return self._sections[section][option]
+        return value
 
 
 class TestPbenchConfig:
-    def mock_file_list(self, cfg_name: str) -> List[str]:
+    @staticmethod
+    def logger_test_setup(monkeypatch, sections: Dict[str, Any]):
+        def mock_ConfigParser_init(self):
+            self._sections = sections
+
+        monkeypatch.setattr(configtools, "file_list", TestPbenchConfig.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_ConfigParser_init)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+
+    def mock_file_list(cfg_name: str) -> List[str]:
         """file_list mock module"""
         return [cfg_name]
 
     def test_default_logger(self, monkeypatch):
-        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
-        monkeypatch.setattr(ConfigParser, "__init__", MockedConfigParser.__init__)
-        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        """Show that PbenchConfig uses default values when NO
+        configuration values are provided for `logging`"""
+
+        self.logger_test_setup(monkeypatch, {})
         config = PbenchConfig("pbench.cfg")
         assert config.logger_type == "devlog"
+        assert config.log_dir is None
+        assert config.log_using_caller_directory is False
+        assert config.default_logging_level == "INFO"
+        assert config.log_fmt is None
+        assert config.TZ == "UTC"
 
-    def test_logger(self, monkeypatch):
-        """verify logger_type value from config file"""
-
-        class mock_MockedConfigParser(MockedConfigParser):
-            def __init__(self):
-                self._sections = {"logging": {"logger_type": "file"}}
-
-        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
-        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
-        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+    def test_custom_logger(self, monkeypatch):
+        """Show that when configuration values are provided,
+        PbenchConfig uses those values instead of default values"""
+
+        logger_type = "file"
+        log_dir = "/mock/log_dir"
+        logging_level = "Devlog"
+        log_format = "1970-01-01T00:00:42.000000 -- 'message'"
+        self.logger_test_setup(
+            monkeypatch,
+            {
+                "logging": {
+                    "logger_type": logger_type,
+                    "log_dir": log_dir,
+                    "logging_level": logging_level,
+                    "log_format": log_format,
+                }
+            },
+        )
         config = PbenchConfig("pbench.cfg")
-        assert config.logger_type == "file"
+        assert config.logger_type == logger_type
+        assert config.log_dir == log_dir
+        assert config.default_logging_level == logging_level
+        assert config.log_fmt == log_format
```

Comment:
This is a "success" case:  should we be testing the whole "interface", here?  (That is, we have four assertions here, whereas at line 55 we have six...should we be checking `log_using_caller_directory` and `TZ` here as well?)

Correspondingly, at line 145, should we be making the other five assertions?

Reply:
Sure. we can add them, but that won't make any difference, as they are hardcoded.

Reply:
> that won't make any difference, as they are hardcoded.

😆 That's true...until someone changes the code!!

Seriously, the reason why the unit test code is separate from the CUT is so that the unit tests can maintain the notion of "truth" regardless of the implementation of the CUT or of subsequent changes to it.

To our best ability, the unit tests should be "black box" tests -- that is, they should be implemented and operate with as little knowledge of how the CUT is implemented as is feasible, while still generating full coverage of the CUT's operation.  (So, yeah, when we write the unit tests, we "peek" inside the CUT, but what the unit tests actually check should be the "external interface" of the "unit", not its internals.)  Thus, the fact that this aspect of the interface happens to be hard-coded today is immaterial:  given that it is part of the interface, it should be checked by the unit tests, and it is appropriate to check it so that a future change to the CUT doesn't break the "interface contract".